### ISSUE TITLE
A3: Question regarding the Negotiate Header field

### ISSUE THREAD

**Issue by tmarshall004** [cs531-f19/discussions#52]
The assignment description specifies that we should send a `300 Multiple Choice` response when the user agent sends `Negotiate: 1.0`.

I have read most of [RFC 2295](https://tools.ietf.org/html/rfc2295#page-28) for the Negotiate header and [RFC 2296](https://tools.ietf.org/html/rfc2296) for specifically rsva 1.0. 
I currently don't understand is precisely what properly supporting rsva 1.0 should change as opposed to it not being specified in the request.
Could you offer some more information or guidance?

**Comment by ibnesayeed** [comment]
RFC 2296 is specifically about the RVSA version 1.0 (i.e., run when `Negotiate: 1.0` is specified). We are not going to implement any other version of RVSA. Presence of this header means we will specifically use the algorithm described in [section 3.3](https://tools.ietf.org/html/rfc2296#section-3.3) to combine various aspects for each available representation to calculate their overall quality and identify the winner. If we still have a tie for the first position, then we will return `300 Multiple Choice` response. Absence of transparent content negotiation related headers means it is either going to be a Proactive or a Reactive negotiation instead. Revisiting Lecture 7 slides might help you understand what I mean.

Let me know if it helps or if you need more information on anything specific.

**Comment by tmarshall004** [comment]
Yes, this does make sense now. Seeing section 3.3 clears up what RSVA 1.0 should be doing.
I assume that currently our servers should be undergoing Reactive negotiation when lacking the `Negotiate` header, always responding with a 300 if there multiple possible resources.
